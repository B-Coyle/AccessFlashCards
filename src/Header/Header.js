import React, {Component} from 'react';
import './Header.scss';

export default class Header extends Component {
    constructor(props) {
        super(props);

    }

    render() {
        return(
            <header className='header'>
                <h1 className="appTitle">Aria Garden</h1>
                <div className='resetContainer'>
                    <input type='button' value='Reset Cards' />
                </div>
                <div className='shuffleContainer'>
                <input type ='button' value='Shuffle Cards' />
                </div>
            </header>
        )
    }
}