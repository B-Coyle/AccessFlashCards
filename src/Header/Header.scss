@import '../Styles/variables.scss';


.header{
    display: grid;
    grid-template-columns: 1fr 1fr 1fr;
    grid-template-areas: 
    "reset title shuffle";
}

.appTitle{
    grid-area: title;
    font-size: 90px;
    text-align: center;
    color: #fff;
    text-decoration: underline;
}

.resetContainer{
    grid-area: reset;
    display: inline;
    text-align:center
}

.resetBtn{
    border-radius: 5px;
    color: $fontColor;
    background-color: green;
    width: 150px;
    height: 50px;
}

.shuffleContainer{
    grid-area: shuffle;
    display: inline;
    text-align:center
}

.resetBtn:hover {
    cursor: pointer;
    background-color: plum;
}