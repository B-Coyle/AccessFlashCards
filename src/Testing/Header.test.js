import React from 'react';
import Header from '../Header/Header';
import { shallow } from 'enzyme'; 
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

const mockResetGame =  jest.fn()
window.addEventListener = jest.fn()

describe('Header', () => {
let wrapper;

beforeEach(()=> {
    wrapper=shallow(
        <Header 
            resetGame = {mockResetGame}
        />
    );
});

it('should match the snapshot with all data passed in', () => {
    expect(wrapper).toMatchSnapshot();
});

it('should have an event listener on the click start button', () => {
    wrapper.find('input').simulate('click')
    expect(mockResetGame).toHaveBeenCalled();

});

})