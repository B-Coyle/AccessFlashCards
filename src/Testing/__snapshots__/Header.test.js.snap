// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot with all data passed in 1`] = `
<header
  className="header"
>
  <h1
    className="appTitle"
  >
    Aria Garden
  </h1>
  <div
    className="resetContainer"
  >
    <input
      onClick={[MockFunction]}
      type="button"
      value="Reset Game from Start"
    />
  </div>
</header>
`;
